noir-projects/noir-contracts/contracts/token_blacklist_contract/src/main.nr

```diff
@@ -214,8 +214,8 @@ contract TokenBlacklist {
         }
 
         let amount = U128::from_integer(amount);
-        storage.balances.sub(from, amount).emit(encode_and_encrypt_note(&mut context, from, from));
-        storage.balances.add(to, amount).emit(encode_and_encrypt_note(&mut context, from, to));
+        storage.balances.sub(from, amount).emit(encode_and_encrypt_note_unconstrained(&mut context, from, from));
+        storage.balances.add(to, amount).emit(encode_and_encrypt_note_unconstrained(&mut context, from, to));
```

Comment:
Did this change because it worked as a workaround around the Noir issue which reoccurred in this function (see [here](https://github.com/noir-lang/noir/issues/5771#issuecomment-2338729198) for context). I didn't want this PR to be blocked no more and we did this change in the standard token so I think it's an ok change to do.

Reply:
Yes, it's a house of cards